app/src/main/kotlin/com/flxrs/dankchat/main/MainFragment.kt

```diff
@@ -403,9 +407,20 @@ class MainFragment : Fragment() {
                         isUserChange = false,
                         loadTwitchData = true,
                     )
-                    val name = dankChatPreferences.userName
-                    if (dankChatPreferences.isLoggedIn && name != null) {
-                        showSnackBar(getString(R.string.snackbar_login, name))
+                    // wait for username to be validated before showing snackbar
+                    collectFlow(dankChatViewModel.oAuthResult) {
+                        when (it) {
+                            is ValidationResult.User      -> showSnackBar(getString(R.string.snackbar_login, it.username))
+                            ValidationResult.TokenInvalid -> {
+                                MaterialAlertDialogBuilder(requireContext())
+                                    .setTitle(getString(R.string.oauth_expired_title))
+                                    .setMessage(getString(R.string.oauth_expired_message))
+                                    .setPositiveButton(getString(R.string.oauth_expired_login_again)) { _, _ -> navigateSafe(R.id.action_mainFragment_to_loginFragment) }
```

Comment:
We should use the existing `openLogin()` method here. Also setting the `isRelogin` parameter to true prevents the cookies from being deleted and in the most cases the login will work without having to input credentials again.